independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/JarFileReference.java

```diff
@@ -223,4 +218,12 @@ private static JarFileReference asyncLoadAcquiredJarFile(JarResource jarResource
         } while (existingJarRef == null || !existingJarRef.acquire());
         return existingJarRef;
     }
+
+    private static JarFileReference completeWith(CompletableFuture<JarFileReference> completableFuture, JarFile jarFile) {
+        Objects.requireNonNull(completableFuture);
+        var jarFileRef = new JarFileReference(jarFile);
+        jarFileRef.completedFuture = completableFuture;
```

Comment:
this should be commented: `jarFileRef.completableFuture` is stored released by `CompletableFuture::complete`